cosmology/GravityManager.h

```diff
@@ -0,0 +1,367 @@
+        , totalP_m(totalP_)
+        , nt_m(nt_)
+        , nr_m(nr_)
+        , lbt_m(lbt_)
+        , solver_m(solver_)
+        , stepMethod_m(stepMethod_){}
+    
+    /**
+     * @brief Destructor for GravityManager.
+     */
+    ~GravityManager(){}
+
+
+public:
+    /**
+     * @brief Folder for initial conditions.
+     */
+    std::string folder;
+    
+
+    /**
+     * @brief Get the total number of particles.
+     * 
+     * @return Total number of particles.
+     */
+    size_type getTotalP() const { return totalP_m; }
+
+    /**
+     * @brief Set the total number of particles.
+     * 
+     * @param totalP_ Total number of particles.
+     */
+    void setTotalP(size_type totalP_) { totalP_m = totalP_; }
+
+    /**
+     * @brief Get the number of time steps.
+     * 
+     * @return Number of time steps.
+     */
+    int getNt() const { return nt_m; }
+
+    /**
+     * @brief Set the number of time steps.
+     * 
+     * @param nt_ Number of time steps.
+     */
+    void setNt(int nt_) { nt_m = nt_; }
+
+    /**
+     * @brief Get the solver type.
+     * 
+     * @return Solver type.
+     */
+    const std::string& getSolver() const { return solver_m; }
+
+    /**
+     * @brief Set the solver type.
+     * 
+     * @param solver_ Solver type.
+     */
+    void setSolver(const std::string& solver_) { solver_m = solver_; }
+
+    /**
+     * @brief Get the load balance threshold.
+     * 
+     * @return Load balance threshold.
+     */
+    double getLoadBalanceThreshold() const { return lbt_m; }
+
+    /**
+     * @brief Set the load balance threshold.
+     * 
+     * @param lbt_ Load balance threshold.
+     */
+    void setLoadBalanceThreshold(double lbt_) { lbt_m = lbt_; }
+
+    /**
+     * @brief Get the step method type.
+     * 
+     * @return Step method type.
+     */
+    const std::string& getStepMethod() const { return stepMethod_m; }
+
+    /**
+     * @brief Set the step method type.
+     * 
+     * @param stepMethod_ Step method type.
+     */
+    void setStepMethod(const std::string& stepMethod_) { stepMethod_m = stepMethod_; }
+
+    /**
+     * @brief Get the number of grid points in each dimension.
+     * 
+     * @return Number of grid points in each dimension.
+     */
+    const Vector_t<int, Dim>& getNr() const { return nr_m; }
+
+    /**
+     * @brief Set the number of grid points in each dimension.
+     * 
+     * @param nr_ Number of grid points in each dimension.
+     */
+    void setNr(const Vector_t<int, Dim>& nr_) { nr_m = nr_; }
+
+    /**
+     * @brief Get the current simulation time.
+     * 
+     * @return Current simulation time.
+     */
+    double getTime() const { return time_m; }
+
+    /**
+     * @brief Set the current simulation time.
+     * 
+     * @param time_ Current simulation time.
+     */
+    void setTime(double time_) { time_m = time_; }
+
+    /**
+     * @brief Set the initial conditions folder.
```

Comment:
Be a bit more descriptive (set the initial conditions folder which is where the ICs that will be read in for the cosmology simulation are)